code/modules/reagents/chemistry/holder.dm

```diff
@@ -348,6 +348,8 @@
   * * liverless - Stops reagents that aren't set as [/datum/reagent/var/self_consuming] from metabolizing
   */
 /datum/reagents/proc/metabolize(mob/living/carbon/C, can_overdose = FALSE, liverless = FALSE)
+	if(C?.dna?.species && (NOREAGENTS in C.dna.species.species_traits))
```

Comment:
mate, we already have a trait that makes you not process reagents (that aren't specifically handled by your race): TRAIT_NOMETABOLISM

look at how skeletons (and how they handle milk) as an example